CMakePresets.json

```diff
@@ -45,8 +45,28 @@
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "RelWithDebInfo"
       }
-    }
-  ],
+    },
+    {
+      "name": "linux-default",
+      "displayName": "Linux Debug",
+      "description": "Sets Ninja generator, compilers, build and install directory, debug build type",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/out/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
```

Comment:
please create the same structure as windows:

a "base" for linux and then per cmake_build_type a target :)

This makes for easy switching between release/debug whilst not having to duplicate the code.